cloud/ec2.go

```diff
@@ -1161,8 +1160,15 @@ func (m *ec2Manager) StopInstance(ctx context.Context, h *host.Host, user string
 		InstanceIds: []*string{aws.String(h.Id)},
 	})
 	if err != nil {
-		if err2 := h.SetStatus(prevStatus, user, ""); err2 != nil {
-			return errors.Wrapf(err2, "failed to revert status from stopping to '%s'", prevStatus)
+		// fetch the instance to get the current status, since it may have changed
+		// since we saved the previous status
+		instance, err = m.client.GetInstanceInfo(ctx, h.Id)
```

Comment:
Can we just call m.GetInstanceStatus? This will also handle translating the status to an Evergreen status

Reply:
I was looking for something like that! thank you!